crypto/cipher_extra/cipher_test.cc

```diff
@@ -256,15 +273,20 @@ static void TestCipherAPI(const EVP_CIPHER *cipher, Operation op, bool padding,
       EVP_CIPHER_CTX_encrypting(ctx.get())) {
     max_out += block_size - (max_out % block_size);
   }
-  std::vector<uint8_t> result(max_out);
+  std::vector<uint8_t> result;
+  max_out ? result.resize(max_out) : result.reserve(1);
```

Comment:
why `reserve(1)` here if `max_out == 0`?

Reply:
This forces an allocation of the vector's underlying array so when you do `result.data()` you get a non-NULL pointer even if `result.size() == 0`. This is important because the cipher interface is really sensitive to the NULL-ness of in/out to activate different steps of the algorithm.